solr/core/src/java/org/apache/solr/cloud/events/ScheduledEvent.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cloud.events;
+
+/**
+ *
+ */
+public interface ScheduledEvent extends ClusterEvent {
+  String getScheduleName();
+  Object getScheduleParam(String key);
```

Comment:
Can/shall we be more specific about what scheduling means? Rather than returning Object that will be cast into...

Reply:
What is a Scheduled event?  

What is a "scheduled-param" ?

Reply:
Thinking more about it, we should use a strongly-typed Schedule.

Scheduled event is something that Solr generates on a predefined schedule ;) It's basically a general-purpose scheduler that is available via this API. There's a need for some periodic maintenance tasks, and currently it was implemented as a part of the autoscaling triggers (ScheduledTrigger, which in turn invoked eg. inactive shard cleanup and collection repair tasks).

I'm open to other suggestions - we could make it a separate API, but modelling it as a cluster event has benefits too (uniform way to register for and to process events, and it avoids adding yet another API).

Reply:
I'm all for these events to be just like the other ones. Just define what a schedule is and strong type whatever can be...

Reply:
> There's a need for some periodic maintenance tasks, and currently it was implemented as a part of the autoscaling triggers

I'm -1 on having the concept of triggers or scheduled events leaking into Solr core. It should be a concept internal to the autoscaling-framework.

Reply:
@chatman even in 8x the scheduled events were used for things other than autoscaling, namely for inactive shard maintenance. Solr still needs this functionality in order to avoid implementing it over and over again in individual plugins.

I don't have a strong opinion whether it must be a part of the event API here, but it seems to me that it's convenient to model scheduled events in this uniform way.